test/js/web/timers/setImmediate.test.js

```diff
@@ -45,3 +47,28 @@ it("clearImmediate", async () => {
   });
   expect(called).toBe(false);
 });
+
+it("setImmediate should not keep the process alive forever", async () => {
+  let process = null;
+  const success = async () => {
+    process = Bun.spawn({
+      cmd: [bunExe(), "run", path.join(import.meta.dir, "process-setImmediate-fixture.js")],
+      stdout: "ignore",
+      env: {
+        ...bunEnv,
+        NODE_ENV: undefined,
+      },
+    });
+    await process.exited;
+    process = null;
+    return true;
+  };
+
+  const fail = async () => {
+    await Bun.sleep(3000);
```

Comment:
lets make this timeout like 500

Reply:
fix in https://github.com/oven-sh/bun/pull/3457/commits/68b783e0bbb8c6721f291fbbfae114197b550491